tests_rv/patch/dtb_warn_rv64/dtb_warn_rv64.sh

```diff
@@ -38,42 +31,32 @@ make -C . O=$tmpdir0 ARCH=riscv CROSS_COMPILE=$CROSS_COMPILE PATH=$PATH \
 	dtbs_check W=1 -j$(nproc) \
 	2> >(tee $tmpfile_o >&2)
 
-incumbent=$(grep -i -c "\(warn\|error\)" $tmpfile_o)
+incumbent=$(cat $tmpfile_o | grep -v "From schema" | wc -l)
 
 echo "Building the tree with the patch"
 
 git checkout -q $HEAD
 
-make -C . O=$tmpdir0 ARCH=riscv CROSS_COMPILE=$CROSS_COMPILE PATH=$PATH \
+make -C . O=$tmpdir1 ARCH=riscv CROSS_COMPILE=$CROSS_COMPILE PATH=$PATH \
 	defconfig
 
-make -C . O=$tmpdir0 ARCH=riscv CROSS_COMPILE=$CROSS_COMPILE PATH=$PATH \
+make -C . O=$tmpdir1 ARCH=riscv CROSS_COMPILE=$CROSS_COMPILE PATH=$PATH \
 	dtbs_check W=1 -j$(nproc) \
 	2> >(tee $tmpfile_n >&2) || rc=1
 
-current=$(grep -i -c "\(warn\|error\)" $tmpfile_n)
+current=$(cat $tmpfile_n | grep -v "From schema" | wc -l)
 
 echo "Errors and warnings before: $incumbent this patch: $current" >&$DESC_FD
 
 if [ $current -gt $incumbent ]; then
   echo "New errors added" 1>&2
+  sed -i 's|^.*arch|arch|g' $tmpfile_o
+  sed -i 's|^.*arch|arch|g' $tmpfile_n
   diff -U 0 $tmpfile_o $tmpfile_n 1>&2
 
-  echo "Per-file breakdown" 1>&2
-  tmpfile_fo=$(mktemp)
-  tmpfile_fn=$(mktemp)
-
-  grep -i "\(warn\|error\)" $tmpfile_o | sed -n 's@\(^\.\./[/a-zA-Z0-9_.-]*.[ch]\):.*@\1@p' | sort | uniq -c \
-    > $tmpfile_fo
-  grep -i "\(warn\|error\)" $tmpfile_n | sed -n 's@\(^\.\./[/a-zA-Z0-9_.-]*.[ch]\):.*@\1@p' | sort | uniq -c \
-    > $tmpfile_fn
-
-  diff -U 0 $tmpfile_fo $tmpfile_fn 1>&2
-  rm $tmpfile_fo $tmpfile_fn
-
   rc=1
 fi
 
-rm -rf $tmpdir0 $tmpfile_o $tmpfile_n
+#rm -rf $tmpdir0 $tmpfile_o $tmpfile_n
```

Comment:
Leaking tempfiles and the new tmpdir1 and tmpdir0? (Are both used?)

Reply:
Woops, I was keeping them locally when I was testing. Also, it's easier to build in two directories than do a clean. I'll add in the second dir and uncomment it.
Thanks.

Reply:
> Woops, I was keeping them locally when I was testing. Also, it's easier to build in two directories than do a clean. I'll add in the second dir and uncomment it. Thanks.

Done, & renamed the directories to match their usage.